File: 884-uncommon-words-from-two-sentences.py
class Solution:
    def uncommonFromSentences(self, s1: str, s2: str) -> List[str]:
        freq = defaultdict(int)

        for word in (s1 + " " + s2).split():
            freq[word] += 1

        return [word for word, count in freq.items() if count == 1]

File: 884-uncommon-words-from-two-sentences.java
class Solution {
    public String[] uncommonFromSentences(String s1, String s2) {
        Map<String, Integer> freq = new HashMap<>();

        String[] words = (s1 + " " + s2).split(" ");

        for (String word : words) {
            freq.put(word, freq.getOrDefault(word, 0) + 1);
        }

        List<String> ans = new ArrayList<>();

        for (Map.Entry<String, Integer> entry : freq.entrySet()) {
            if (entry.getValue() == 1) {
                ans.add(entry.getKey());
            }
        }

        return ans.toArray(new String[0]);
        
    }
}
